pkg/channelmultiplexer/channelmultiplexer.go

```diff
@@ -7,22 +7,37 @@ import (
 	"github.com/stackrox/rox/pkg/sync"
 )
 
+// OptionFunc provides options for the ChannelMultiplexer.
+type OptionFunc[K any] func(*ChannelMultiplexer[K])
+
+// WithChannel provides a context to the ChannelMultiplexer,
+// If it's not provided context.Background will be used.
+func WithChannel[K any](ctx context.Context) OptionFunc[K] {
```

Comment:
Why not `WithContext`?

Reply:
Yeah... I don't know that I was thinking 😅